### PR TITLE
Add a github action to sync shared source from aspnetcore

### PR DESCRIPTION
**Pull request by Tratcher**
This mirrors an action we have in aspnetcore to sync changes to shared HTTP source code files.
https://github.com/dotnet/aspnetcore/actions/workflows/runtime-sync.yml

The one in aspnet triggers on a nightly schedule and sends a PR to aspnetcore if it detects any diffs. However, sometimes changes need to be submitted to runtime instead. The new action added here will only trigger manually, and will send a pr to runtime with any diffs.

### CODE REVIEW

File: .github/workflows/aspnetcore-sync.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+
+jobs:
+  compare_repos:
+    # Comment out this line to test the scripts in a fork
+    if: github.repository == 'dotnet/runtime'
+    name: Compare the shared code in the AspNetCore and Runtime repos and send a PR to Runtime if they're out of sync.
```
Comment (safern) [inline .github/workflows/aspnetcore-sync.yml:15]: Seems like a long name for an action 😆 -- maybe we can shorten it a little bit?

**Comment by JamesNK** [comment]
A couple of questions:

* How is the sync triggered?
* How are reaction changes on the PR added? For example, a sync change renames a method and runtime code needs to react to that.

File: .github/workflows/aspnetcore-sync.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: windows-latest
+    steps:
+    - name: Checkout aspnetcore
+      uses: actions/checkout@v2.0.0
+      with:
+        # Test this script using changes in a fork
+        repository: 'dotnet/aspnetcore'
+        # repository: 'tratcher/aspnetcore'
```
Comment (safern) [inline .github/workflows/aspnetcore-sync.yml:23]: Maybe we can remove these commented repositories?

File: .github/workflows/aspnetcore-sync.yml (hunk 1 above)
Comment (Tratcher) [reply .github/workflows/aspnetcore-sync.yml:15]:
```suggestion
    name: Sync Code
```

File: .github/workflows/aspnetcore-sync.yml (hunk 2 above)
Comment (Tratcher) [reply .github/workflows/aspnetcore-sync.yml:23]:
```suggestion
```

File: .github/workflows/aspnetcore-sync.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        path: aspnetcore
+        ref: main
+    - name: Checkout runtime
+      uses: actions/checkout@v2.0.0
+      with:
+        # Test this script using changes in a fork
+        repository: 'dotnet/runtime'
+        # repository: 'tratcher/runtime'
```
Comment (Tratcher) [inline .github/workflows/aspnetcore-sync.yml:31]:
```suggestion
```

**Comment by Tratcher** [comment]
> A couple of questions:
> 
> * How is the sync triggered?

It's triggered manually from the https://github.com/dotnet/runtime/actions page. Compare to this one that has a "Run Workflow" option: https://github.com/dotnet/runtime/actions/workflows/create-codespaces-prebuild.yml

> * How are reaction changes on the PR added? For example, a sync change renames a method and runtime code needs to react to that.

You'd check out the branch and push additional changes.

**Review by JamesNK — APPROVED** [review]
I don't know anything about github actions.

This approve is moral support for the feature 😄

**Comment by Tratcher** [comment]
FYI, here is an example of the issue and PR the runtime->aspnetcore version of this script generates:
https://github.com/dotnet/aspnetcore/issues/18943#issuecomment-977624821
https://github.com/dotnet/aspnetcore/pull/38614

This is also a case where the changes need to go into runtime instead. The new aspnetcore->runtime script only generates a PR in runtime, it references the same issue in aspnetcore.

File: .github/workflows/aspnetcore-sync-checkdiff.ps1 [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Check the code is in sync
+$changed = (select-string "nothing to commit" artifacts\status.txt).count -eq 0
```
Comment (safern) [inline .github/workflows/aspnetcore-sync-checkdiff.ps1:2]: Is this extra file needed? We could just inline this command in the gh action?

Also, I wonder if using `git diff` would be easier? Something like

```
git diff -M -C -b --ignore-cr-at-eol --ignore-space-at-eol --exit-code > ..\artifacts\diff.txt
$changed = !$?
```

The only issue I see with `git diff` are untracked files, so we would need to run `git add -N .` before running the command. Just an idea to consider.

Comment (Tratcher) [reply .github/workflows/aspnetcore-sync-checkdiff.ps1:2]: I guess the aspnetcore version of this file did a lot more to open and comment on the github issue. Inlining just this part might be easier now.
https://github.com/dotnet/aspnetcore/blob/main/.github/workflows/ReportDiff.ps1

**Review by ManickaP — APPROVED** [review]
Love the idea, thanks for submitting this!

**Comment by Tratcher** [comment]
I'm going to merge this so we can do a live test, and then iterate on feedback.

**Pull request merged**